libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -571,7 +579,22 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 			else
 				args.emplace_back(convert(*arguments[i], *parameterTypes[i]).commaSeparatedList());
 
-		if (auto identifier = dynamic_cast<Identifier const*>(&_functionCall.expression()))
+		if (auto memberAccess = dynamic_cast<MemberAccess const*>(&_functionCall.expression()))
```

Comment:
This does not have to be a library call - you can also call base functions without virtual semantics by using `BaseContractName.functionName` - the code (minus the assertions) would work for both.

Reply:
Good point. The base calls are not implemented yet, right? I'll modify the assertions and also add `solUnimplementedAssert` for those calls.

Reply:
They will be if you remove the assertion about `functionDef->virtualSemantics` :)

Reply:
Assertions removed. Now the code works for both libraries and base contracts.